test/files/run/t10513.scala

```diff
@@ -0,0 +1,34 @@
+//package scala.concurrent
+import scala.concurrent._
+
+import scala.util.{Random, Try}
+import ExecutionContext.Implicits.global
+
+/** This test uses recursive calls to Future.flatMap to create arrays whose
+ *  combined size is slightly greater than the JVM heap size. A previous
+ *  implementation of Future.flatMap would retain references to each array,
+ *  resulting in a speedy OutOfMemoryError. Now, each array should be freed soon
+ *  after it is created and the test should complete without problems.
+ */
+object Test {
+ val arrSz = 50 * 10000
+  val numFutures = 10000
+
+  val rng = new Random()
```

Comment:
Why are these outside of `main`?

Reply:
first two are simply constants, rng can be either static class member or local variable, can't really see how it matters for a test class being executed just once :smile: .
I actually recall now, this used to be a scalatest suite with one failing test and a second one with the fix implemented inline, so it made sense to stay DRY.

Reply:
I'd prefer if they were contained within `main` :)